cmd/run.go

```diff
@@ -60,11 +69,29 @@ type runConfig struct {
 
 	// Verbose logging.
 	Verbose bool
+
+	EnvVars map[string]string
 }
 
-func newRunConfig() runConfig {
+func newRunConfig(cmd *cobra.Command) runConfig {
+	envVarsM := make(map[string]string)
+
+	if cmd.Flags().Changed("env") {
+		envVarsA, err := cmd.Flags().GetStringArray("env")
+		if err == nil {
+			for _, s := range envVarsA {
+				kvp := strings.Split(s, "=")
+				if len(kvp) == 2 && kvp[0] != "" {
+					envVarsM[kvp[0]] = kvp[1]
+				} else if len(kvp) == 1 &&  kvp[0] != "" {
+					envVarsM[kvp[0]] = ""
+				}
+			}
+		}
+	}
```

Comment:
Just at a quick glance this looks duplicated from above. Could it be extracted into a function?